config/routes.rb

```diff
@@ -193,11 +193,19 @@
         put :update_mentor, path: "update-mentor"
 
         collection do
-          get :add, to: "add_participants/base#start"
-
-          namespace :add_participants, path: "add" do
-            resource :type, only: %i[show update]
-            resource :details, only: %i[show update]
+          get :add, to: "add_participants#start"
+
+          scope(
+            controller: "add_participants",
+            path: "add",
+            constraints: {
+              step: Regexp.union(
+                *Schools::AddParticipantForm::STEPS.map { |step| step.to_s.tr("_", "-") },
+              ),
+            },
+          ) do
+            get ":step", action: :show
+            post ":step", action: :update
```

Comment:
How many steps do we expect to have there?

If it's less than 5 I think having a pair of `get` and `put/post` endpoints is a solution that takes similar amount of space, but a lot less brain power to process...

Reply:
Something between 6 or 9 depending on decision if certain scenarios display the same step with conditionals or completely different steps.

I agree this might seem a bit complicated, but as befere - this code extracts a technical code in order to make business logic easier to follow. Ideally I'd extract this later into a separate file(s), reducing the whole controller into just few configuration lines.